plugin/src/main/java/org/opensearch/sql/plugin/SQLPlugin.java

```diff
@@ -202,6 +201,7 @@ public Collection<Object> createComponents(
     dataSourceService.createDataSource(defaultOpenSearchDataSourceMetadata());
     LocalClusterState.state().setClusterService(clusterService);
     LocalClusterState.state().setPluginSettings((OpenSearchSettings) pluginSettings);
+    LocalClusterState.state().setClient(client);
```

Comment:
"LocalClusterState" is a misnomer now that it is just a pass-through to making a call via the Client, though renaming might make this change more intrusive.

Reply:
Agreed!